src/library/process.js

```diff
@@ -56,77 +46,125 @@ const getCurrentUserData = async (AWS, userId) => {
  * @param {string} service is the name of the service
  * @param { string } account is AWS the account number
  * @param { msgBody: object, msgAttribs: object, rcptHandle: string } is the message object containing properties msgBody (Body of the SNS message), msgAttribs (Message attributes of the SNS message) and rcptHandle (Message receipt handle for the message from SQS)
- * @param {Function} msgHandler is the handler function that will proess the message
+ * @param {Function} msgHandler is the handler function that will process the message
  * @returns {Boolean}
  * @throws {Error}
  */
-export const processMessage = async (AWS, region, service, account,
-  { msgBody, msgAttribs, rcptHandle }, msgHandler) => {
+export const processMessage = async (
+  AWS,
+  region,
+  service,
+  account,
+  { msgBody, msgAttribs, rcptHandle },
+  msgHandler,
+) => {
   // Use the neverThrowError function from utils to process the message making
   // sure that no error is thrown back
 
-  const dbConfig = await fetchRecordsByQuery(
-    AWS,
-    {
-      TableName: 'vendorConfig',
-      ExpressionAttributeNames: { '#pk': 'service' },
-      KeyConditionExpression: '#pk = :serv',
-      ExpressionAttributeValues: {
-        ':serv': { S: `${service}` },
-      },
+  const dbConfig = await fetchRecordsByQuery(AWS, {
+    TableName: 'vendorConfig',
+    ExpressionAttributeNames: { '#pk': 'service' },
+    KeyConditionExpression: '#pk = :serv',
+    ExpressionAttributeValues: {
+      ':serv': { S: `${service}` },
     },
-  );
+  });
 
   // eslint-disable-next-line no-undef-init
   let serviceAccountData = {};
-  if (itemExists(msgBody, 'context')
-    && itemExists(msgBody.context, 'user')
-    && itemExists(msgBody.context.user, 'accountId')) {
-    const accData = await getCurrentAccountData(
-      AWS, msgBody.context.user.accountId,
-    );
-    if (itemExists(accData, 'vendorData')
-      && itemExists(accData.vendorData, `${service}`)) {
+  if (
+    itemExists(msgBody, 'context') &&
+    itemExists(msgBody.context, 'user') &&
+    itemExists(msgBody.context.user, 'accountId')
+  ) {
+    const accData = await getCurrentAccountData(AWS, msgBody.context.user.accountId);
+    if (itemExists(accData, 'vendorData') && itemExists(accData.vendorData, `${service}`)) {
       serviceAccountData = accData.vendorData[`${service}`];
     }
+    if (!itemExists(msgBody.context, account)) {
+      msgBody.context.account = accData;
+    }
+  }
+
+  // eslint-disable-next-line
+  let sharedMicroApplicationAccountData = {};
+  if (
+    itemExists(msgBody, 'context') &&
+    itemExists(msgBody.context, 'user') &&
+    itemExists(msgBody.context.user, 'accountId')
+  ) {
+    const accData = await getCurrentAccountData(AWS, msgBody.context.user.accountId);
+    if (itemExists(accData, 'sharedMicroApplicationAccountData')) {
```

Comment:
My thinking, was to name these variables consistently with what the actual data store would be called ( currently "vendorData" ), however idk, it may be harder to understand this way.

Reply:
That way in the actually Dynamo entities it would looks like this:

account: {
  "sharedMicroApplicationAccountData": {
    "antivirus": {
     "microApplicationsToShareWith": ['*'],
      "serviceData": {
        "key" : "value"
      }
    }
  }
}

again though... definitely lengthy, I will brainstorm on how to make this more intuitive.